### PR TITLE
Update readme for v1.0.0

### PR DESCRIPTION
**Pull request by MFraters**
I noticed that I hadn't updated the readme. Need to add that to the release checklist.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ For more information see the [GWB site](https://geodynamicworldbuilder.github.io
 
 ## Frequently Asked Questions (FAQ)
 ### What are the minimum requirements to compile the library and tools?
-For GWB version 0.6.0 only a C++14 compiler and CMake version higher than 2.8.12 are required to compile the library tools. Note that for some optional features (such as certain wrappers and unity build) a higher version of CMake and other dependencies may be needed (e.g., a Fortran compiler to compile the Fortran wrapper).
+For GWB version 1.0.0 only a C++14 compiler and CMake version higher than 2.8.12 are required to compile the library tools. Note that for some optional features (such as certain wrappers and unity build) a higher version of CMake and other dependencies may be needed (e.g., a Fortran compiler to compile the Fortran wrapper).
```
Comment (gassmoeller) [inline README.md:24]: Make your life easier by mentioning the specific number in as few places as possible:
```suggestion
Currently, GWB requires only a C++14 compiler and CMake version higher than 2.8.12 to compile the library tools. Note that for some optional features (such as certain wrappers and unity build) a higher version of CMake and other dependencies may be needed (e.g., a Fortran compiler to compile the Fortran wrapper).
```

File: README.md [hunk 2]
```diff
@@ -36,11 +36,13 @@ Feel free to join!
 ### How to cite?
 The developers of the Geodynamic World Builder request that you cite the following publication:
 
+Fraters, M., Billen, M., Gassmöller, R., Saxena, A., Heister, T., Li, H., Douglas, D., Dannberg J., Bangerth W. and Wang, Y.: The Geodynamic World Builder: A planetary structure creator for the geosciences, Journal of Open Source Software, [https://doi.org/10.21105/joss.06671](https://doi.org/10.21105/joss.06671), 2024.
+
 Fraters, M., Thieulot, C., van den Berg, A., and Spakman, W.: The Geodynamic World Builder: a solution for complex initial conditions in numerical modelling, Solid Earth, [https://doi.org/10.5194/se-10-1785-2019](https://doi.org/10.5194/se-10-1785-2019), 2019.
 
-And cite the specific version of the software used. Version 0.6.0 can be cited as:
+And cite the specific version of the software used. Version 1.0.0 can be cited as:
```
Comment (gassmoeller) [inline README.md:43]:
```suggestion
And cite the specific version of the software used. The current release can be cited as:
```

**Review by gassmoeller — APPROVED** [review]
Either merge as is, or include my suggestions to save yourself some work in the future ;-)

**Comment by MFraters** [comment]
haha, that is a good idea. I took your suggestions :)

**Pull request merged**